### ISSUE TITLE
Validation of optional fields referring to choices inappropriately fails

### ISSUE THREAD

**Issue by whimboo** [anweiss/cddl#139]
Trying to validate the following JSON file against the given CDDL definition fails. As it looks like there is a problem with the optional occurrence indicator (including `*`, `+` as well) because removing it will succeed the validation.

```JSON
{
  "name": "foo",
  "valid": "no"
}
```

```CDDL
Argument = {
  name: text,
  ?valid: "yes" / "no"
}
```

The visible failure:
```
[ERROR] Validation of "data.json" failed: error validating type choice at the root of the JSON document: unexpected key "valid"
```

Removing the question mark it succeeds:
```
[INFO] Validation of "data.json" is successful
```

**Comment by anweiss** [comment]
Thanks for reporting @whimboo. Oversight on my part when implementing https://github.com/anweiss/cddl/pull/138.

**Comment by whimboo** [comment]
> Thanks for reporting @whimboo. Oversight on my part when implementing #138.

You are welcome. Interesting for me is that #138 landed post 0.9.0 right and I can see the issue with that given release.
